src/eos-kolibri-export

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+
+# Copyright © 2023 Endless OS Foundation, LLC
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or (at
+# your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+from argparse import ArgumentParser
+from gi.repository import Gio
+import logging
+import os
+import re
+import shlex
+import shutil
+import subprocess
+import time
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def system_server_proxy():
+    bus = Gio.bus_get_sync(Gio.BusType.SYSTEM, None)
+    return Gio.DBusProxy.new_sync(
+        bus,
+        Gio.DBusProxyFlags.NONE,
+        None,  # interface info
+        'org.learningequality.Kolibri.Daemon',
+        '/org/learningequality/Kolibri/Daemon/Main',
+        'org.learningequality.Kolibri.Daemon',
+    )
+
+
+def get_system_kolibri_home(timeout=10):
+    deadline = time.time() + timeout
+    while True:
+        proxy = system_server_proxy()
+        homedir = proxy.get_cached_property('KolibriHome').unpack()
+        logger.debug(f'System Kolibri homedir: {homedir}')
+        if homedir != '':
+            return homedir
+        if time.time() >= deadline:
+            raise Exception(
+                f'Cound not determine system Kolibri homedir in {timeout} '
+                'seconds'
+            )
+        time.sleep(0.5)
+
+
+def run(cmd, check=True, **kwargs):
+    cmdstr = shlex.join(cmd)
+    logger.debug(f'> {cmdstr}')
+    return subprocess.run(cmd, check=check, **kwargs)
+
+
+def filter_logs(output):
+    log_pattern = re.compile('^(CRITICAL|ERROR|WARNING|INFO|DEBUG)')
+    for line in output.splitlines(keepends=True):
+        if ord(line[0]) == 0o33:
+            continue
+        if log_pattern.match(line):
+            continue
+        yield line
```

Comment:
That's so weird we need to do this! I feel like this must have regressed in Kolibri some time in the past year, since our documentation around the listcontent script happily suggests piping stdout to a file and we definitely tested it at the time: https://support.endlessos.org/en/deployment/image-builder/kolibri. We should definitely try to find where this broke.

Reply:
Indeed, quite sad. I thought I put in a comment linking to the issue (or really any comments), but I guess not. It's been happening since 0.15.2, I believe.

https://github.com/learningequality/kolibri/issues/9162
https://github.com/learningequality/kolibri/pull/9240
https://github.com/learningequality/kolibri/issues/10297

I thought about adding a method to save the file from listcontent, but then it can only be saved somewhere accessible in the sandbox and it would be owned by the wrong user. And then it would depend on getting new releases out and I wanted to make this work with existing releases if possible.

Another thought I had was to monkeypatch the Kolibri logging setup code for listcontent to return it to sanity.